src/webapi/Controllers/EvaluationRatingController.cs

```diff
@@ -83,6 +83,7 @@ public async Task<IActionResult> Post(PerformedEvaluationResult evaluationResult
                     newEvalRating.PersonId = newObjRating.PersonId = evaluationResult.ReviewedPersonId;
                     newEvalRating.SourceSystemDescriptor = newObjRating.SourceSystemDescriptor = evaluationResult.ReviewedPersonSourceSystemDescriptor;
                     newEvalRating.UserId = newObjRating.UserId = user.Id;
+                    newEvalRating.CandidateName = evaluationResult.ReviewedPersonName;
```

Comment:
@stevenarnold-ed-fi, CandidateName is already on Candidate. Why adding it here directly if it can be got from Candidate with PersonId and SourceSystemDescriptor?

Reply:
Convenience - on the main page we show completed evaluations, including the name of the candidate. If we don't save the candidate's name locally we would need to make calls to the ODS/API to retrieve candidates for just their names. This will also be needed in ETA-42 where we want to load a previously performed evaluation for review purposes
![image](https://github.com/Ed-Fi-Exchange-OSS/EvaluationTracker/assets/48261255/8b0be717-5478-4ba8-9149-1029f8c64535)

Reply:
Ok, so this has exposed an issue. In EvaluationController.GetEvaluationObjectivesElementsTitles(...) we use PerformanceEvaluations to create AvailableEvaluationObjectives that are sent to the front-end. I had already added the ReviewedCandidateName to it to "cache" CandidateName. However, upon saving, we're creating/updating EvaluationRatings, EvaluationObjectives and EvaluationElementRatingResults. In short, we're reading from a source and writing in a different one.
This leads me to the need of refactoring to sync these sources. In the interest of locally syncing as little as possible we can build "virtual" dependencies for objects high in the hierarchy from ones lower in it if their non-PK columns are unused. For example, we can "create" a PerformanceEvaluation from an EvaluationElementRatingResult if we don't need PerformanceEvaluation.PerformanceEvaluationDescription. Any dependency in which we use a non-PK field must be synced locally (e.g: EvaluationObjectiveRating because of Comments).
Additionally, we should locally sync Candidates to avoid the online GET every time, which would also allow us to save CandidateName in the right model.